### PR TITLE
[Java] extract concepts from gigasecond example

### PR DESCRIPTION
**Pull request by msomji**

### CODE REVIEW

**Comment by iHiD** [comment]
Does this also use loops or conditionals (`if`) or anything like that?

**Comment by pdmoore** [comment]
Should the gigasecond.md file be in languages/java/reference/exercise-concepts/ directory? Asking before I create one for Leap.

**Comment by ErikSchierboom** [comment]
@pdmoore Well spotted! It should indeed be located in `languages/java/reference/exercise-concepts/gigasecond.md`.

**Comment by ErikSchierboom** [comment]
Very nice! I'll leave the more in-depth, Java specific review to the Java maintainers

File: languages/java/reference/exercise-concepts/gigasecond.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+- Assignment: assigning values, in this case, the private property that contains the date
+
+- Properties: In object oriented programming classes tend to have state, such state is maintained within properties to be set/read at a later time during runtime
+
+- Overloading constructors: Sometimes there can be multiple ways to initialize a class, this is known as overloading since the name of the methods match the name of the class but only vary by their arguments.
+
+- Scoping: `{` and `}` used to identify scope of method 
+
+- Assiginment: assigining values to private properties
```
Comment (mirkoperillo) [inline languages/java/reference/exercise-concepts/gigasecond.md]: Is this a repetition of line 15 ?

Comment (msomji) [reply languages/java/reference/exercise-concepts/gigasecond.md]: Yes! removed

**Comment by msomji** [comment]
@iHiD Sorry i missed your comment, but no, there are no `if` loops in this one, overloading constructors seems to be the most straightforward approach we'd expect/want students to take.

**Comment by pdmoore** [comment]
Noticed a typo on line 31 - "deonted"

**Pull request merged**